src/IOOption.ts

```diff
@@ -0,0 +1,728 @@
+} from './FromEither'
+import { chainFirstIOK as chainFirstIOK_, chainIOK as chainIOK_, FromIO1, fromIOK as fromIOK_ } from './FromIO'
+import { flow, identity, Lazy, pipe, SK } from './function'
+import { bindTo as bindTo_, flap as flap_, Functor1 } from './Functor'
+import * as _ from './internal'
+import { Monad1 } from './Monad'
+import { MonadIO1 } from './MonadIO'
+import { NaturalTransformation11, NaturalTransformation21 } from './NaturalTransformation'
+import * as O from './Option'
+import * as OT from './OptionT'
+import { Pointed1 } from './Pointed'
+import { Predicate } from './Predicate'
+import { ReadonlyNonEmptyArray } from './ReadonlyNonEmptyArray'
+import { Refinement } from './Refinement'
+import { Separated } from './Separated'
+import * as I from './IO'
+import { URI as IEURI } from './IOEither'
+import { Zero1, guard as guard_ } from './Zero'
+
+import IO = I.IO
+import Option = O.Option
+
+// -------------------------------------------------------------------------------------
+// model
+// -------------------------------------------------------------------------------------
+
+/**
+ * @category model
+ * @since 2.12.0
+ */
+export interface IOOption<A> extends IO<Option<A>> {}
+
+// -------------------------------------------------------------------------------------
+// constructors
+// -------------------------------------------------------------------------------------
+
+/**
+ * @category constructors
+ * @since 2.12.0
+ */
+export const some: <A>(a: A) => IOOption<A> =
+  /*#__PURE__*/
+  OT.some(I.Pointed)
+
+/**
+ * @category constructors
+ * @since 2.12.0
+ */
+export const fromPredicate: {
+  <A, B extends A>(refinement: Refinement<A, B>): (a: A) => IOOption<B>
+  <A>(predicate: Predicate<A>): <B extends A>(b: B) => IOOption<B>
+  <A>(predicate: Predicate<A>): (a: A) => IOOption<A>
+} =
+  /*#__PURE__*/
+  OT.fromPredicate(I.Pointed)
+
+// -------------------------------------------------------------------------------------
+// natural transformations
+// -------------------------------------------------------------------------------------
+
+/**
+ * @category natural transformations
+ * @since 2.12.0
+ */
+export const fromOption: NaturalTransformation11<O.URI, URI> = I.of
+
+/**
+ * @category natural transformations
+ * @since 2.12.0
+ */
+export const fromEither: FromEither1<URI>['fromEither'] =
+  /*#__PURE__*/
+  OT.fromEither(I.Pointed)
+
+/**
+ * @category natural transformations
+ * @since 2.12.0
+ */
+export const fromIO: FromIO1<URI>['fromIO'] =
+  /*#__PURE__*/
+  OT.fromF(I.Functor)
+
+/**
+ * @category natural transformations
+ * @since 2.12.0
+ */
+export const fromIOEither: NaturalTransformation21<IEURI, URI> =
+  /*#__PURE__*/
+  I.map(O.fromEither)
+
+// -------------------------------------------------------------------------------------
+// destructors
+// -------------------------------------------------------------------------------------
+
+/**
+ * @category destructors
```

Comment:
would a `fold` make sense?

Reply:
@gkamperis in all other modules `fold` is just an alias of `match` (or `matchE`) now

Reply:
@gcanti yes saw that... but since it existed in `IOEither` thought I'd mention it to see if we should keep the pattern...